src/routers/alpha-router/alpha-router.ts

```diff
@@ -1214,20 +1236,60 @@ export class AlphaRouter
     return trade;
   }
 
-  private buildMethodParameters(
+  private async buildMethodParameters(
```

Comment:
Can we separate these into 2 methods? Maybe: `buildMethodParametersForSwap` `buildMethodParametersForSwapAndAdd`